internal/persistence/persistors/mysqlstore/store_organization.go

```diff
@@ -0,0 +1,312 @@
+		}
+
+		if len(filters.OrganizationNameIn) > 0 {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.Name.IN(filters.OrganizationNameIn))
+		}
+
+		if filters.CreatedBy.Valid {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.CreatedBy.EQ(filters.CreatedBy))
+		}
+
+		if filters.LastUpdatedBy.Valid {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.LastUpdatedBy.EQ(filters.LastUpdatedBy))
+		}
+	}
+
+	q := mysqlmodel.Organizations(queryMods...)
+	totalCount, err := q.Count(ctx, ctxExec)
+	if err != nil {
+		return nil, fmt.Errorf("get organizations count: %w", err)
+	}
+
+	page := pagination.Page
+	maxRows := pagination.MaxRows
+	if filters != nil {
+		if filters.Page.Valid {
+			page = filters.Page.Int
+		}
+		if filters.MaxRows.Valid {
+			maxRows = filters.MaxRows.Int
+		}
+	}
+
+	pagination.SetQueryBoundaries(page, maxRows, int(totalCount))
+
+	queryMods = append(queryMods, qm.Limit(pagination.MaxRows), qm.Offset(pagination.Offset))
+	q = mysqlmodel.Organizations(queryMods...)
+
+	if err = q.Bind(ctx, ctxExec, &res); err != nil {
+		return nil, fmt.Errorf("get organizations: %w", err)
+	}
+
+	fmt.Println("the querymods --- ", queryMods)
+
+	pagination.RowCount = len(res)
+	paginated.Organizations = res
+	paginated.Pagination = pagination
+
+	return &paginated, nil
+}
+
+func (m *Repository) CreateOrganization(ctx context.Context, tx persistence.TransactionHandler, organization *model.Organization) (*model.Organization, error) {
+	ctxExec, err := mysqltx.GetCtxExecutor(tx)
+	if err != nil {
+		return nil, fmt.Errorf("extract context executor: %w", err)
+	}
+
+	entry := mysqlmodel.Organization{
+		Name: organization.Name,
+	}
+	if err = entry.Insert(ctx, ctxExec, boil.Infer()); err != nil {
+		return nil, fmt.Errorf("insert organization: %w", err)
+	}
+
+	organization, err = m.GetOrganizationById(ctx, tx, entry.ID)
+	if err != nil {
+		return nil, fmt.Errorf("get organization by id: %w", err)
+	}
+
+	return organization, nil
+}
+
+func (m *Repository) ConvertNameToID(ctx context.Context, tx persistence.TransactionHandler, userName string) (int, error) {
+	ctxExec, err := mysqltx.GetCtxExecutor(tx)
+	if err != nil {
+		return 0, fmt.Errorf("extract context executor: %w", err)
+	}
+
+	var userID int
+	err = ctxExec.QueryRowContext(ctx, "SELECT id FROM users WHERE name = ?", userName).Scan(&userID)
+	if err != nil {
+		return 0, fmt.Errorf("query user id: %w", err)
+	}
+	return userID, nil
+}
+
+func (m *Repository) AddOrganization(ctx context.Context, tx persistence.TransactionHandler, organization *model.Organization) (*model.Organization, error) {
+	ctxExec, err := mysqltx.GetCtxExecutor(tx)
+	if err != nil {
+		return nil, fmt.Errorf("extract context executor: %w", err)
+	}
+
+	createdByID, err := m.ConvertNameToID(ctx, tx, organization.CreatedBy)
+	if err != nil {
+		return nil, fmt.Errorf("convert createdBy to ID: %w", err)
+	}
```

Comment:
AddOrganization(ctx context.Context, tx persistence.TransactionHandler, organization *model.Organization) (*model.Organization, error)

to

AddOrganization(ctx context.Context, tx persistence.TransactionHandler, userId int, organization *model.Organization) (*model.Organization, error)

Reply:
I actually have provided [details for this](https://github.com/dembygenesis/theoverwatchtools.cli/pull/33/files#r1705923075), but it was lost in the duplicate. Better trim that out.